cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -426,6 +426,14 @@ pub fn deregister_activation(
     Ok(current_activations)
 }
 
+/// Returns whether this watchdog has been started without a need for it
+/// e.g. another watchdog is already monitoring this activation.
+pub fn should_bail_at_startup(reg: &EnvRegistry, path_hash: &str) -> bool {
+    reg.entry_for_hash(path_hash)
+        .map(|entry| !entry.activations.is_empty())
```

Comment:
Will this be a partially breaking change for existing activations of an environment with services? I'm thinking of a situation where:
1. activation1 with old `flox activate -s` starts watchdog1 and services
2. activation2 with old `flox activate` starts watchdog2
3. activation3 with new `flox activate` doesn't start watchdog
4. activation1 exits and watchdog1 exits without cleaning up
5. activation2 exist and watchdog2 exits and cleans up services
6. activation3 no longer has services

I don't have any immediate suggestions about how to codify around it and I'd rather we start using the code paths than develop everything on a long running branch, so maybe it's an acceptable trade-off if we document it in the release notes?

Reply:
As written yes, I agree it would be a breaking change if you have running activations across the upgrade and want to create another instance of that activation. I think it's something we put in the release notes because it doesn't feel like a _huge_ imposition to say that you need to close all of your currently running activations before doing the upgrade (I think).

Reply:
In general we're okay with telling people they need to exit old activations. We have https://github.com/flox/flox/issues/2162 to detect that and tell users to do it. I think it's probably okay if we only communicate that in release notes this once. Although in a best case scenario, we may be able to do 2162 before the next release